r/tools/nixlibs.R

```diff
@@ -328,20 +328,25 @@ build_libarrow <- function(src_dir, dst_dir) {
     # CXXFLAGS = R_CMD_config("CXX11FLAGS"), # We don't want the same debug symbols
     AR = R_CMD_config("AR"),
     RANLIB = R_CMD_config("RANLIB"),
-    LDFLAGS = R_CMD_config("LDFLAGS"),
-    CMAKE_CXX_COMPILER_AR = R_CMD_config("LDFLAGS")
+    LDFLAGS = R_CMD_config("LDFLAGS")
   )
   env_vars <- paste0(names(env_var_list), '="', env_var_list, '"', collapse = " ")
   env_vars <- with_s3_support(env_vars)
   env_vars <- with_mimalloc(env_vars)
-  if (tolower(Sys.info()[["sysname"]]) %in% "sunos") {
+  if (tolower(Sys.info()[["sysname"]]) %in% "sunos" || grepl("-flto", env_var_list[["LDFLAGS"]])) {
```

Comment:
If you wanted to be slightly more targeted you could also check for `"-flto"` in `R CMD config LTO`

Reply:
Maybe, but I think this is where it matters, this is what gets used in the link step right?

Reply:
Hmm, `-flto` doesn't show up in LDFLAGS: `LDFLAGS="-L/usr/local/lib"`